keel-sql/src/main/resources/db/changelog/20191108-environment-artifact-constraint.yml

```diff
@@ -0,0 +1,71 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-environment-artifact-constraint-table
+      author: asher
+      changes:
+        - createTable:
+            tableName: environment_artifact_constraint
+            columns:
+              - column:
+                  name: environment_uid
+                  type: char(26)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: artifact_version
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: constraint_type
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: status
+                  type: int
+                  constraints:
+                    nullable: true
+              - column:
+                  name: judged_by
+                  type: varchar(255)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: judged_at
+                  type: timestamp
+                  constraints:
+                    nullable: true
+              - column:
+                  name: comment
+                  type: varchar(255)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: json
```

Comment:
Intended for storing `constraint_type` specific attributes. A manual judgement constraint likely wouldn't use this at all but a canary constraint may store metatdata about a canary run such as kayenta id, score, report link, etc.

Reply:
Should this be `details` or something like that vs labeled `json`?

Reply:
I don't love `json` but noticed there's a column of that name on the `resource_event` table. The `resource` table has a `metatdata` longtext column for serialized blobs too though, how about if I change this to that?

Reply:
oh, i'd say just leave it as `json` then like the `resource_event` table. I like that more than `metadata` because I feel like `metadata` has a specific meaning.